js_modules/dagster-ui/packages/ui-core/src/app/BaseFallthroughRoot.tsx

```diff
@@ -42,29 +39,40 @@ const FinalRedirectOrLoadingRoot = () => {
     return <Redirect to="/locations" />;
   }
 
-  const reposWithVisibleJobs = allRepos.filter((r) => getVisibleJobs(r).length > 0);
+  // If there are visible jobs, redirect to overview
+  const anyVisibleJobs = ({repository}: DagsterRepoOption) => {
+    return repository.pipelines.some(({name}) => !isHiddenAssetGroupJob(name));
+  };
+  const anyReposWithVisibleJobs = allRepos.some((r) => anyVisibleJobs(r));
+  if (anyReposWithVisibleJobs) {
+    return <Redirect to="/overview" />;
+  }
 
-  // If we have no repos with jobs, see if we have an asset group and route to it.
-  if (reposWithVisibleJobs.length === 0) {
-    const repoWithAssetGroup = allRepos.find((r) => r.repository.assetGroups.length);
-    if (repoWithAssetGroup) {
-      return (
-        <Redirect
-          to={workspacePath(
-            repoWithAssetGroup.repository.name,
-            repoWithAssetGroup.repositoryLocation.name,
-            // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-            `/asset-groups/${repoWithAssetGroup.repository.assetGroups[0]!.groupName}`,
-          )}
-        />
-      );
-    }
+  // If there are jobs but they are all hidden, route to the overview timeline grouped by automation.
+  const hasAnyJobs = allRepos.some((r) => r.repository.pipelines.length > 0);
+  if (hasAnyJobs) {
+    return <Redirect to="/overview/activity/timeline?groupBy=automation" />;
   }
-  if (reposWithVisibleJobs.length > 0) {
-    return <Redirect to="/overview" />;
+
+  // If we have no repos with jobs, see if we have an asset group and route to it.
+  const repoWithAssetGroup = allRepos.find((r) => r.repository.assetGroups[0]);
+  if (repoWithAssetGroup) {
+    const {repository, repositoryLocation} = repoWithAssetGroup;
+    const assetGroup = repository.assetGroups[0];
```

Comment:
TS will still regard this as being maybe-undefined. :(

Reply:
Hm, `yarn ts` is clean – is there another check to do here?

Reply:
Ah I was running the wrong command. Fixing now.